client/engine/chainservice/eth_chainservice.go

```diff
@@ -71,14 +72,15 @@ func NewEthChainService(chain ethChain, na *NitroAdjudicator.NitroAdjudicator,
 	logger := zerolog.New(logDestination).With().Timestamp().Str("txSigner", txSigner.From.String()[0:8]).Caller().Logger()
 
 	// Use a buffered channel so we don't have to worry about blocking on writing to the channel.
-	ecs := EthChainService{chain, na, naAddress, caAddress, vpaAddress, txSigner, make(chan Event, 10), logger}
+	ecs := EthChainService{chain, na, naAddress, caAddress, vpaAddress, txSigner, make(chan Event, 10), logger, make(chan struct{})}
 
 	if ecs.subscriptionsSupported() {
 		logger.Printf("Notifications are supported by the chain. Using notifications to listen for events.")
-		go ecs.subscribeForLogs(context.Background())
+		context, cancel := context.WithCancel(context.Background())
+		go ecs.subscribeForLogs(context, cancel)
```

Comment:
Is this the right way to use `context.WithCancel`?   I thought the `cancel` function is generally called by the context constructor/provider, and it triggers  [Done](https://pkg.go.dev/context#Context.Done) on the context consumers.

I think the way this is implemented now the call to `cancel` by `ecs.subscribeForLogs` will do nothing as there is nothing that actually makes use of the context.

Reply:
The context is threaded all the way through to the `ecs.chain.SubscribeFilterLogs(ctx, query, logs)` call. Although `ecs.subscribeForLogs` never reads `ctx.Done`, I think `chain.SubscribeFilterLogs` should?

This may be more confusing than necessary. It might be clearer / better to construct the `context` _inside_ of `ecs.subscribeForLogs`?

Reply:
4ada56158196440d656ab3c2590c9060b2813912